### PR TITLE
MB-14227: independent session managers for each app

### PR DESCRIPTION
**Pull request by ahobson**
## Summary

Instead of having a single session store for for all apps, create separate stores for each. This ensures we can't accidentally have session collisions or accidentally find the wrong session for the wrong app.

In addition, refactor the session managers to use a struct instead of an array.

## Setup to Run Your Code

There should be no visible changes to how the app works. There are new assertions in the session_test that ensures independent session stores.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)

### CODE REVIEW

File: pkg/handlers/apitests.go [hunk 1]
```diff
@@ -25,6 +23,29 @@ import (
...
+	PrimeTestHost string = "prime.example.com"
+)
+
+// ApplicationTestServername is a collection of the test servernames
+func ApplicationTestServername() auth.ApplicationServername {
+	return auth.ApplicationServername{
+		MilServername:    MilTestHost,
+		OfficeServername: OfficeTestHost,
+		OrdersServername: OrdersTestHost,
+		AdminServername:  AdminTestHost,
```
Comment (mboren) [inline pkg/handlers/apitests.go:45]: is `PrimeTestHost` intentionally missing here?

File: pkg/handlers/apitests.go [hunk 2]
```diff
@@ -46,27 +67,16 @@ func NewBaseHandlerTestSuite(sender notifications.NotificationSender, packageNam
...
-	adminSession.Cookie.Name = "admin_session_token"
-
-	officeSession = scs.New()
-	officeSession.Store = store
-	officeSession.Cookie.Name = "office_session_token"
-
-	return [3]*scs.SessionManager{milSession, adminSession, officeSession}
+func setupSessionManagers() auth.AppSessionManagers {
+	return auth.SetupSessionManagers(nil, false, time.Duration(180*time.Second),
+		time.Duration(180*time.Second))
```
Comment (mboren) [inline pkg/handlers/apitests.go:79]: extreme nitpick: I think wrapping with `time.Duration(...)` here is redundant.
```suggestion
		180*time.Second
```

File: pkg/handlers/authentication/devlocal.go [hunk 3]
```diff
@@ -959,7 +955,26 @@ func createSession(h devlocalAuthHandler, user *models.User, userType string, w
...
+	// ctx, err := sessionManager.Load(r.Context(), session.IDToken)
+	// if err != nil {
+	// 	appCtx.Logger().Error("Cannot load session from context", zap.Error(err))
+	// 	return nil, err
+	// }
+	// _, _, err = sessionManager.Commit(ctx)
+	// if err != nil {
+	// 	appCtx.Logger().Error("Cannot commit session to context", zap.Error(err))
+	// 	return nil, err
+	// }
```
Comment (mboren) [inline pkg/handlers/authentication/devlocal.go:975]: Is this still useful? I don't totally understand what this snippet would do for us when debugging.

**Comment by mboren** [comment]
I think this is looking good, I only found a couple of small things I had questions about. I will test on experimental later today

File: pkg/handlers/apitests.go (hunk 2 above)
Comment (ahobson) [reply pkg/handlers/apitests.go:79]: Thank you!

File: pkg/handlers/authentication/devlocal.go (hunk 3 above)
Comment (ahobson) [reply pkg/handlers/authentication/devlocal.go]: No, it's not! I've removed it

File: pkg/handlers/apitests.go (hunk 1 above)
Comment (ahobson) [reply pkg/handlers/apitests.go:45]: 😊 Nope!

**Comment by mboren** [comment]
Tested on experimental by logging into each of the apps with login.gov accounts simultaneously, and logging in and out a few times. Didn't observe any strange behavior with sessions.

**Review by mboren — APPROVED** [review]

**Pull request merged**